Api/CalcWrapper.cs

```diff
@@ -0,0 +1,298 @@
+                if (String.IsNullOrEmpty(Path))
+                {
+                    xComponent = aLoader.loadComponentFromURL("private:factory/scalc", "_blank", 0, loadProperties);
+                }
+                else
+                {
+                    try
+                    {
+                        Path = String.Format("file:///{0}", Path);
+                        xComponent = aLoader.loadComponentFromURL(Path, "_blank", 0, loadProperties);
+                    }
+                    catch (unoidl.com.sun.star.uno.Exception ex)
+                    {
+                        throw new unoidl.com.sun.star.uno.Exception(ex.Message, ex);
+                    }
+                }
+            }
+            else
+            {
+                loadProperties[0].Name = "Hidden";
+                loadProperties[0].Value = new uno.Any(false);
+                if (String.IsNullOrEmpty(Path))
+                {
+                    xComponent = aLoader.loadComponentFromURL("private:factory/scalc", "_blank", 0, new unoidl.com.sun.star.beans.PropertyValue[0]);
+                }
+                else
+                {
+                    try
+                    {
+                        Path = String.Format("file:///{0}", Path);
+                        xComponent = aLoader.loadComponentFromURL(Path, "_blank", 0, new unoidl.com.sun.star.beans.PropertyValue[0]);
+                    }
+                    catch (unoidl.com.sun.star.uno.Exception ex)
+                    {
+                        throw new unoidl.com.sun.star.uno.Exception(ex.Message, ex);
+                    }
+                }
+            }
+            return (unoidl.com.sun.star.sheet.XSpreadsheetDocument)xComponent;
+        }
+
+        private void SaveDocument(String Path)
+        {
+            try
+            {
+                Path = Path.Replace("\\", "/"); // Convert forward slashes to backslashes, converting it to the correct format storeToURL expects. 
+                Path = String.Concat("file:///", Path);
+                XStorable xStorable = (XStorable)mxDocument; // Typecast the currently open document to XStorable type.
+                xStorable.storeToURL(Path, new unoidl.com.sun.star.beans.PropertyValue[1]); //Creating an empty PropertyValue array saves the document in the default .ods format.
+            }
+            catch(unoidl.com.sun.star.uno.Exception ex)
+            {
+                throw new unoidl.com.sun.star.uno.Exception(ex.Message, ex);
+            }
+        }
+
+        private void AddNewSheet(String Name)
+        {
+            try
+            {
+                unoidl.com.sun.star.sheet.XSpreadsheets xSheets = mxDocument.getSheets();
+                short sheetIndex = Convert.ToInt16(xSheets.getElementNames().Length + 1); //Get the number of sheets already existing in the document, add one to get the new index. 
```

Comment:
Remove the comment and move the code to separate method with a name that explains what it does:
```csharp
        private void AddNewSheet(string name)
        {
            var xSheets = MxDocument.getSheets();
            var sheetIndex = GetCountOfExistingSheets(xSheets) + 1;
            xSheets.insertNewByName(name, (short)sheetIndex);
        }

        private static int GetCountOfExistingSheets(XSpreadsheets xSheets)
        {
            return xSheets.getElementNames().Length;
        }
```